### PR TITLE
[BugFix] Fix query information_schema.be_logs with incorrect log limit parameter

### PR DESCRIPTION
**Pull request by sduzh**
…t parameter

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by decster — APPROVED** [review]

**Comment by kevincai** [comment]
need fix the PR title and commit message.

part of the title words wrapped into commit message.

File: fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java [hunk 1]
```diff
@@ -1194,7 +1194,7 @@ public PlanFragment visitPhysicalSchemaScan(OptExpression optExpression, ExecPla
             }
 
             // set a per node log scan limit to prevent BE/CN OOM
-            if (scanNode.getLimit() > 0) {
+            if (scanNode.getLimit() > 0 && predicates.isEmpty()) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java:1197]: any context for this change?

Comment (decster) [reply fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java:1197]: If the query has predicates, should not push down limit to scannode

Comment (decster) [reply fe/fe-core/src/main/java/com/starrocks/sql/plan/PlanFragmentBuilder.java:1197]: The logic here is oversimplified, but it is practical. There are some conditions that the limit should be push down, i.e. 
```
select log from be_logs where log rlike 'xxx'
```
If all predicate is also pushed down, we can push down limit, but the logic to check all predicates are pushed down is complex, so it's not implemented.

**Comment by decster** [comment]
> need fix the PR title and commit message.
> 
> part of the title words wrapped into commit message.

fixed

**Review by kevincai — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**